### PR TITLE
Periodically cleanup the onDemand:members sets if the referenced keys…

### PR DESCRIPTION
**Pull request by ajordens**
… no longer exist

- keys are being TTL'd out so this just keeps the onDemand:members  set in-sync
- runs as part of a new core provider

### CODE REVIEW

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/core/agent/CleanupPendingOnDemandCachesAgent.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  }
+
+  @Override
+  public String getProviderName() {
+    return CoreProvider.PROVIDER_NAME;
+  }
+
+  @Override
+  public void run() {
+    getCatsModule().getProviderRegistry().getProviders().forEach(provider -> {
```
Comment (ajordens) [inline clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/core/agent/CleanupPendingOnDemandCachesAgent.java]: Specifically this aims to cleanup the onDemand:members sets for Aws/AwsInfrastructure/Titus.

File: clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/core/agent/CleanupPendingOnDemandCachesAgent.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+  public long getPollIntervalMillis() {
+    return pollIntervalMillis;
+  }
+
+  public long getTimeoutMillis() {
+    return timeoutMillis;
+  }
+
+  private Set<String> scanMembers(String setKey) {
```
Comment (ajordens) [inline clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/core/agent/CleanupPendingOnDemandCachesAgent.java]: This was lifted from `RedisCache`.

**Comment by ajordens** [comment]
PTAL @cfieber -- I've had this sitting locally for a little while but it provides a bit of house keeping around the `onDemand:members` sets now that the underlying keys TTL out after 10 minutes.

Again this only handles scenarios where the normal evict after edda read process doesn't kick in.

**Comment by cfieber** [comment]
LGTM

**Pull request merged**